// @mgcrea/pino-pretty-compact/dist/index.cjs
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  colorBoolean: () => colorBoolean,
  colorDate: () => colorDate,
  colorJson: () => colorJson,
  colorKeyword: () => colorKeyword,
  colorNumber: () => colorNumber,
  colorString: () => colorString,
  colorStringArray: () => colorStringArray,
  default: () => src_default
});
module.exports = __toCommonJS(src_exports);

// src/utils/color.ts
var color = __toESM(require("kolorist"), 1);
var colorString = (s) => color.green(`'${s}'`);
var colorKeyword = (s) => color.blue(`${s}`);
var colorJson = (s) => color.green(JSON.stringify(s));
var colorStringArray = (a) => `[ ${a.map(colorString).join(", ")} ]`;
var colorNumber = (n) => color.yellow(`${n}`);
var colorBoolean = (b) => color.yellow(b ? "true" : "false");
var colorDate = (d) => color.magenta(d.toISOString());

// src/prettifier.ts
var color3 = __toESM(require("kolorist"), 1);
var import_os2 = require("os");
var import_pino_pretty = __toESM(require("pino-pretty"), 1);

// src/utils/error.ts
var isObject = (maybeObject) => typeof maybeObject === "object" && maybeObject !== null;
var isSerializedError = (maybeError) => {
  return isObject(maybeError) && !!maybeError["message"];
};
var serializeError = (message) => {
  const raw = new Error(String(message));
  return {
    type: "Error",
    message: String(message),
    stack: String(raw.stack),
    raw
  };
};

// src/utils/format.ts
var color2 = __toESM(require("kolorist"), 1);

// src/config/static.ts
var { OS_HOSTNAME } = process.env;
var LOG_LEVEL_LABEL = {
  [10 /* TRACE */]: "trace",
  [20 /* DEBUG */]: "debug",
  [30 /* INFO */]: " info",
  [40 /* WARN */]: " warn",
  [50 /* ERROR */]: "error",
  [60 /* FATAL */]: "fatal"
};

// src/utils/format.ts
var import_os = require("os");
var CWD = process.cwd();
var CWD_REGEX = new RegExp(CWD, "g");
var colorForLevel = (level) => {
  switch (level) {
    case 10 /* TRACE */:
      return color2.gray;
    case 20 /* DEBUG */:
      return color2.cyan;
    case 30 /* INFO */:
      return color2.green;
    case 40 /* WARN */:
      return color2.yellow;
    case 50 /* ERROR */:
      return (...args) => color2.bold(color2.red(...args));
    case 60 /* FATAL */:
      return (...args) => color2.bold(color2.bgRed(...args));
    default:
      return color2.white;
  }
};
var colorMsgForLevel = (level) => {
  switch (level) {
    case 10 /* TRACE */:
      return color2.gray;
    case 20 /* DEBUG */:
      return color2.white;
    case 30 /* INFO */:
      return color2.white;
    case 40 /* WARN */:
      return color2.yellow;
    case 50 /* ERROR */:
      return (...args) => color2.bold(color2.red(...args));
    case 60 /* FATAL */:
      return (...args) => color2.bold(color2.bgRed(...args));
    default:
      return color2.white;
  }
};
var formatLevel = (level) => colorForLevel(level)(LOG_LEVEL_LABEL[level]);
var formatProcessId = (pid) => color2.magenta(`*${pid}`);
var formatHostname = (hostname) => color2.gray(`@${hostname}`);
var formatSessionId = (id) => color2.magenta(`%${id}`);
var formatRequestId = (id) => color2.magenta(`#${id}`);
var formatPlugin = (plugin) => color2.gray(`(${plugin})`);
var formatErrorStack = (stack) => color2.gray(stack.replace(CWD_REGEX, ".").split(import_os.EOL).slice(1).join(import_os.EOL));
var formatError = (error, level) => {
  const { statusCode = 500, type = error["name"], stack = `${import_os.EOL}    at ???` } = error;
  const supportsArt = color2.options.supportLevel === 2;
  const icon = supportsArt ? "\xD7" : "x";
  const isInternalError = !statusCode || statusCode >= 500;
  const output = [color2[isInternalError ? "red" : "yellow"](`${icon}${type} `), color2.magenta(statusCode)];
  if (isInternalError) {
    output.push(colorForLevel(level)(`: ${error.message}`), import_os.EOL, formatErrorStack(stack));
  } else {
    output.push(`: ${error.message}`);
  }
  return output.join("");
};

// src/prettifier.ts
var defaultOptions = {
  ignore: "pid,hostname",
  colorize: color3.options.enabled,
  errorLikeObjectKeys: ["error", "err"],
  singleLine: true,
  hideObject: true,
  translateTime: "yyyy-mm-dd'T'HH:MM:sso"
};
var prettifyTime = (inputData) => color3.gray(String(inputData));
var build = (options3) => {
  const {
    errorLikeObjectKeys = defaultOptions.errorLikeObjectKeys,
    ignore = defaultOptions.ignore,
    colorize = defaultOptions.colorize
  } = options3;
  const ignoredKeys = ignore.split(",");
  if (colorize && !color3.options.enabled) {
    color3.options.enabled = true;
    color3.options.supportLevel = 2;
  }
  const messageFormat = (log, messageKey, _leveLabel) => {
    const { level, time, msg, reqId, sessionId, plugin, silent, ...otherProps } = log;
    if (silent) {
      return "";
    }
    const output = [];
    if (!ignoredKeys.includes("reqId") && reqId) {
      output.push(formatRequestId(reqId), " ");
    }
    if (!ignoredKeys.includes("sessionId") && sessionId) {
      output.push(formatSessionId(sessionId), " ");
    }
    const firstErrorKey = errorLikeObjectKeys.find((key) => log[key]);
    const formattedMsg = colorMsgForLevel(level)(String(log[messageKey]));
    if (firstErrorKey) {
      const error = log[firstErrorKey];
      const serializedError = isSerializedError(error) ? error : serializeError(error);
      output.push(formattedMsg, import_os2.EOL, " ", formatError(serializedError, level), import_os2.EOL);
    } else {
      output.push(formattedMsg);
    }
    if (!ignoredKeys.includes("plugin") && plugin) {
      output.push(" ", formatPlugin(plugin));
    }
    const outputProps = Object.keys(otherProps).reduce((soFar, key) => {
      if (errorLikeObjectKeys.includes(key) || ignoredKeys.includes(key)) {
        return soFar;
      }
      soFar[key] = otherProps[key];
      return soFar;
    }, {});
    if (Object.keys(outputProps).length > 0) {
      output.push(" ", colorJson(outputProps));
    }
    return output.concat(import_os2.EOL).join("");
  };
  return (0, import_pino_pretty.default)({
    ...defaultOptions,
    customPrettifiers: {
      time: prettifyTime,
      level: formatLevel,
      hostname: formatHostname,
      pid: formatProcessId
    },
    messageFormat,
    ...options3,
    colorize: false
  });
};

// src/index.ts
var src_default = build;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  colorBoolean,
  colorDate,
  colorJson,
  colorKeyword,
  colorNumber,
  colorString,
  colorStringArray
});
